compiler/src/generate/t_ruby_generator.cc

```diff
@@ -809,52 +825,79 @@ void t_ruby_generator::generate_field_data(
     t_type* field_type,
     const std::string& field_name = "",
     t_const_value* field_value = NULL,
-    bool optional = false) {
+    bool optional = false,
+    std::map<string, string>* annotations = NULL) {
```

Comment:
double check - this `annotation` param may be NULL on some code path? If it is always non-NULL, safe to use reference `std::map<string, string>&` instead.

Reply:
Yes, it gets called with `NULL` [here a few times](https://github.com/airbnb/sparsam/blob/1788ee097ddfa0d6f226d03636be094fe081d7d1/compiler/src/generate/t_ruby_generator.cc#L854-L865).

@liangg @andyfangdz thank you so much for reviewing!  Does this mean we can merge it 😄